modules/node_groups/node_groups.tf

```diff
@@ -15,8 +15,9 @@ resource "aws_eks_node_group" "workers" {
 
   ami_type        = lookup(each.value, "ami_type", null)
   disk_size       = lookup(each.value, "disk_size", null)
-  instance_types  = each.value["launch_template_id"] != null ? [] : [each.value["instance_type"]]
+  instance_types  = each.value["launch_template_id"] != "" ? [] : flatten([each.value["instance_type"]])
```

Comment:
What was the reason that the check was changed from verifying if the `launch_template_id` is `""` instead of `null`? I forked and tried this PR and the resulting `instance_types` was a `t3.large`. I didn't have a `launch_template_id` anywhere (`workers_group_defaults`, `node_group_defaults` or `node_groups`).

Changing the predicate back to `each.value["launch_template_id"] != null` resulted in the `instance_types` being picked up from the `instance_type` set in `var.node_groups`.

Reply:
Thanks @duboisf. I don't recall exactly how, but at least on terraform 0.14, null doesn't work for me, whereas "" does.
What's your version of Terraform? Should I be testing in 0.12 instead?

Reply:
It might be a side effect of https://github.com/terraform-aws-modules/terraform-aws-eks/commit/571da60aea10034cef20c435c2b49933ab5ede47

Reply:
Ok, I understand now. It's indeed related to the PR I mention above. I was doing my tests by editing files directly in my .terraform folder and forgot to update the value of launch_template_id to null in local.tf. I reverted that value back to null and pushed a new commit. Please check now.

Reply:
Tested it, it works :+1: